第4回課題提出

Aose0924 opened the pull request.

Aose0924 committed:
第四回課題

Aose0924 committed:
Update lecture04.md

dai-fuji reviewed (commented):
以下確認お願いします。
- 画像もリポジトリにプッシュして相対パスで表示できるように調整お願いします！
- RDS構築部分の画像がEC2の起動画面になっていますので確認お願いします！
- ECのセキュリティグループでSSHポートを全開にしているようなので、ご自身のIPに絞るようお願いします。

Aose0924 committed:
第四回課題キャプチャ

Aose0924 committed:
Add files via upload

Aose0924 committed:
Add files via upload

Aose0924 committed:
Add files via upload

Aose0924 committed:
Add files via upload

Aose0924 committed:
ご指摘箇所の修正

Aose0924 commented:
ご指摘箇所の修正を行いました。
再三で恐れ入りますが、内容のご確認をお願いします。

dai-fuji commented on EC2_SSH制御_0414-6.PNG:
アウトバウンドルールですが、下側のルールの中に上側のルールが内包されています。
重複しているルールの存在は管理が複雑になるため、重複のないようにお願いします！

Aose0924 replied:
すみません。
アウトバウンドルールで内包されているとご指摘の箇所は、以下2つのことかと認識しています。認識相違ないですか？
raisetech-kensyo-SSHはSSH用で外すと接続できなくなるかと思います。
ec2-rds-1はOracle用で外すとOracleへ接続できなくなります。

raisetech-kensyo-SSH	sg-01c179691e82d32d6
ec2-rds-1	sg-01b9a8e4b28887651

dai-fuji replied:
失礼しました。
改めて見たところ特に問題ありませんでした。
他の方課題提出と混同していたようです。

dai-fuji reviewed (commented):
- RDSのセキュリティグループの設定状態が分かるエビデンスを追加お願いします。
- RDSのサブネットグループがプライベートサブネットに限定されているかが分かるエビデンスを追加提出お願いします。

Aose0924 committed:
ご指摘箇所エビデンス

Aose0924 committed:
ご指摘箇所追加

Aose0924 commented:
ご指摘箇所の修正を行いました。
再三で恐れ入りますが、内容のご確認をお願いします。

dai-fuji commented on RDS_セキュリティグループ設定_0416-1.PNG:
VPCのリソースマップを拝見するとプライベートサブネットは2つのようですが、サブネットグループには3つのサブネットが含まれているようです。
ご確認お願いします。

Aose0924 replied:
ご返信ありがとうございます。
すみません、ちょっとご質問の意味が良く解っていないのですが、以下確認をさせてください。
・サブネットグループに3つのサブネットが含まれていて何が問題でしょうか。
意図しないサブネットが含まれると、意図しないIPからの接続を許可してしまうから問題という認識でよいですか？
・サブネットグループに3つのサブネットが含まれているとのことですが、どこを見られておられますか
![image](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/0b3c049d-c469-4b22-8578-e42195d074cc)

dai-fuji replied:
> ・サブネットグループに3つのサブネットが含まれていて何が問題でしょうか。
> 意図しないサブネットが含まれると、意図しないIPからの接続を許可してしまうから問題という認識でよいですか？

概ねその通りです！
多くの場合、DBはインターネット経由でのアクセスを必要とせず、またアクセスされないよう制限するためにIGWへのルートを持たないプライベートサブネットに配置します。

また、講座資料としてはパブリックサブネットとプライベートの説明のところで、どんなものがどちらのサブネットに配置されるものか解説がありますね！

なお、RDSを利用した構成はWEB上にもたくさん転がっています。AWSの公式のチュートリアルやリファレンスアーキテクチャなどで調べてみるといいと思います！

> ・サブネットグループに3つのサブネットが含まれているとのことですが、どこを見られておられますか

このコメントをやりとりしている対象の画像を見ています。
このコメントは特定のGitHubの機能を利用して画像ファイルに対して紐付く形で投稿していますが、それはご認識ありますでしょうか？
その画像を見てもらうと分かると思うのですが。

dai-fuji replied:
補足)リソースマップはマークダウンで利用しているこちらを参照しています！
VPC_0407-8.PNG

Aose0924 replied:
ご回答ありがとうございます。
以下のように修正したのですが、意図とあっておりますでしょうか。
RDSのサブネットグループは、default-vpc-0b4f2dec9fece8099と認識しています。
そのサブネットグループに属するサブネットは、以下３つと認識しています。
subnet-09a19edf403a4564f
subnet-0a99ea00a5e4abc2c
subnet-0c799557b3bf6a13d

上記サブネットは、全て172.16.xxx.xxx/20になっていて重複していたために、subnet-09a19edf403a4564fとsubnet-0c799557b3bf6a13dは削除しました。
subnet-0a99ea00a5e4abc2cの172.16.0.0/20のみ残しています。
![ご質問-RDS_セキュリティグループ設定_0418-1](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/2d24da14-0716-4377-8805-a085c76dafcb)
![ご質問-RDS_セキュリティグループ設定_0418-2](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/693bdd53-d4b5-4bab-abd3-0c8133ff73dc)
![ご質問-RDS_セキュリティグループ設定_0418-3](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/aef57179-677c-4c9c-968c-abfb551097c0)
![ご質問-RDS_セキュリティグループ設定_0418-4](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/49e3570d-388e-466a-8ab5-d10e33db7420)

dai-fuji commented on RDS_プライベートサブネット確認_0416-2.PNG:
こちらはVPCの画面のため、サブネットの確認はできません。

Aose0924 replied:
すみません、こちらも良く解らないのですが、
VPCID語尾4桁8099のサブネットは、16ビット(172.31.0.0/16)のサブネットということをお伝えしたかったのですが、、RDSの画面でサブネットを確認する画面があるということでしょうか？
RDSからサブネットの記載がある画面を抽出してきたのですが以下画面のことですか
![image](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/8484e2bd-d8fa-41ab-90bf-b4d27fbce767)

dai-fuji replied:
このコメントに紐付く画像を見る限り、まずVPCを確認するための画面のようなので、サブネットがそもそも確認できません。。。
なので最初のコメント内容になっています。

要求は以下です。
> RDSのサブネットグループがプライベートサブネットに限定されているか

そのために必要なものは、
- RDSのサブネットグループにはどのサブネットが含まれているか
- それらのサブネットがプライベートサブネットであるかどうか。
これらを示していただきたいです！

また、一点確認したいのですがVPCは10点台か172点台どちらで作成されたのでしょうか？
マークダウンに関連づけられている画像は10点台なのですが、コメントに貼り付けていただいている画像では172点台なようなので、実際に作成いただいているものと、報告内容が違うのかな？と感じました。

Aose0924 replied:
頓珍漢な回答でしたらすみません。
RDSのサブネットグループは、以下の画像からdefault-vpc-0b4f2dec9fece8099と認識しています。
そのサブネットグループに属するサブネットは、以下と認識しています。
subnet-09a19edf403a4564f
subnet-0a99ea00a5e4abc2c
subnet-0c799557b3bf6a13d

上記サブネットは、全て172.16.xxx.xxx/20になっていて重複していたために、subnet-09a19edf403a4564fとsubnet-0c799557b3bf6a13dは削除しました。
subnet-0a99ea00a5e4abc2cの172.16.0.0/20のみ含めております。
![ご質問-RDS_セキュリティグループ設定_0418-1](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/abbbcb50-cc2d-40e7-be6f-ffb33ce712e9)
![ご質問-RDS_セキュリティグループ設定_0418-2](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/39edbe4a-2395-4f55-abd2-35d6ae5de4f5)
![ご質問-RDS_セキュリティグループ設定_0418-3](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/e0c69e24-93fb-4814-ab86-38a3aac75b52)
![ご質問-RDS_セキュリティグループ設定_0418-4](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/ee06aede-736f-4a23-aa98-3c520df61a25)

dai-fuji replied:
> 上記サブネットは、全て172.16.xxx.xxx/20になっていて重複していたために、subnet-09a19edf403a4564fとsubnet-0c799557b3bf6a13dは削除しました。
> subnet-0a99ea00a5e4abc2cの172.16.0.0/20のみ含めております。

こちらどういう修正意図でしょうか？
「CIDRが重複しているから２つのサブネットを削除した」とありますが、当初私からサブネットグループに関して言及していたのは、「サブネットグループに含まれているサブネットがプライベートサブネットか」です。

Aose0924 replied:
３つのプライベートサブネットがあって、それぞれのサブネットのCIDRのプライベートIPの範囲が重複していると思ったので1つに統一したというのが修正の意図です。
すみません、下記でもご説明していますが、VPCを整理しましたので再度ご確認いただけないでしょうか。

dai-fuji replied:
同一のVPCでIPアドレスの重複があるサブネットは仕様上作成できないので、おそらく何か勘違いされていると思います。

dai-fuji reviewed (commented):
コメントバックしましたので確認お願いします。

Aose0924 commented:
すみません。上記の通りご質問をさせていただきます。
お手数ですがご回答いただいてもよいでしょうか？

dai-fuji commented:
先のコメントにも記載しましたが、マークダウンおよび、コメントのやり取りで複数のVPCが登場しており、そもそもこの課題で構築するリソースをどのVPCに構築されようとしているのか、が良く分からない状態になっていますので、まずここを明確にするよる必要があるかと思います。
VPCについて整理お願いできますか？

Aose0924 commented:
RDSとEC2インスタンスを紐づけるVPCが複数あり混乱したため、1つのVPC（vpc-041b64c68c22b15c7）に統一しました。
CIDRは10.0.0.0/16になります。
※VPC
![RDS_0418-1-1](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/e7007342-88b8-4277-9edb-bb1547e33cb4)
※RDS
![RDS_0418-1](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/94db8f69-d34a-44d0-a6b0-e813cc24fae8)
※EC2
![RDS_0418-1-2](https://github.com/tatsuyaaose/KENSYO_REPO/assets/25246044/32e67941-a03b-4a97-ab4e-1995aa2ba75a)

dai-fuji commented:
lecture04.mdに見直し内容を反映させて再度ご連絡いただけますか？

Aose0924 committed:
Update lecture04.md

Aose0924 committed:
Add files via upload

Aose0924 committed:
Add files via upload

Aose0924 committed:
Add files via upload

Aose0924 committed:
Update lecture04.md

Aose0924 commented:
lecture04.mdに見直し内容を反映しました。
再三で恐縮ですが、ご確認をお願いします。

dai-fuji commented on lecture04.md:
```diff
@@ -0,0 +1,92 @@
+![RDS_0409-1-2](/RDS_0409-1-2.PNG)
+![RDS_0409-1-3](/RDS_0409-1-3.PNG)
+![RDS_0409-1-4](/RDS_0409-1-4.PNG)
+![RDS_0409-1-5](/RDS_0409-1-5.PNG)
```
見出しの内容と合っていない画像が含まれていると思いますので確認お願いします。

dai-fuji commented on lecture04.md:
```diff
@@ -0,0 +1,92 @@
+![RDS_0409-1-4](/RDS_0409-1-4.PNG)
+![RDS_0409-1-5](/RDS_0409-1-5.PNG)
+#### VPCをraisetech-vpc (vpc-041b64c68c22b15c7)へ変更
+![RDS_0418-1-6](/RDS_0418-1-6.PNG)
```
> RDSのサブネットグループがプライベートサブネットに限定されているか

こちら確認お願いします！

dai-fuji reviewed (commented):
コメントしました！

Aose0924 committed:
Add files via upload

Aose0924 committed:
Add files via upload

Aose0924 committed:
Update lecture04.md

Aose0924 commented:
lecture04.mdに見直し内容を反映しました。
再三で恐縮ですが、ご確認をお願いします。

dai-fuji commented on lecture04.md:
```diff
@@ -34,36 +35,32 @@
+#### EC2インスタンスをRDSに接続
+![RDS_0419-1-1](/RDS_0419-1-1.PNG)
+![RDS_0419-1-2](/RDS_0419-1-2.PNG)
+#### RDSのサブネットグループがプライベートサブネットに限定されていることを確認
```
エビデンスを見たところプライベートサブネットに限定されていないと思いますが、どのように考えられているか教えていただけますと！

Aose0924 replied:
RDS_0419-2.PNGの画面で、サブネットグループ内のサブネットIDに属するCIDRブロックが、プライベートIPのみ割り振られているので問題ないと考えていました。
CIDRブロックを見て、プライベートかパブリックかを判断していたのですが、この考え方は間違いでしょうか？

dai-fuji replied:
ちょっと違いますね！

プライベートサブネットとパブリックサブネットの定義を一度学習された方がよいですね！！

Aose0924 replied:
承知しました。
お手数ですが参考になるようなURLはございますか？
プライベートサブネットとパブリックサブネットの定義は以下のように認識していました。
・パブリックサブネット：
インターネットGWに接続するようにルートテーブルに定義されたサブネット、つまりネット接続できるサブネット
・プライベートサブネット：
デフォルトGWがインターネットGWに接続されていないサブネット、NATの仕組みがあればネット接続できるけれど、それ以外はネット接続できないサブネット

dai-fuji replied:
プライベートサブネットとパブリックサブネットの認識はバッチリです！
先ほどのコメントにプライベートipが割り振られてあるかどうかというような記載がありましたが、今回説明していただいた定義をもとに改めて確認してみて下さい！
分からなければまたコメント下さい！

Aose0924 replied:
すみません。確認しましたがよくわからないです。
アベイラビリティゾーンやCIDRブロックのプライベートIPが重複しているのが問題なのでしょうか…
参考となるURLなど共有いただけましたら幸いです

dai-fuji replied:
既に必要な情報は集まっているので、閲覧すべきサイトはありませんね！
また、AZの重複は関係ありません！そして先のコメントでもお伝えしましたが同一VPCでサブネットのCIDRの重複は仕様上発生することはないので、勘違いされているか他のVPCのサブネットを参照されているかと！

次の手順で確認してみて下さい！
1.サブネットグループに含まれているサブネットはどのサブネットか、またいくつあるか確認。

2.それらのサブネットがパブリックサブネットかプライベートサブネットかを確認。(関連づけられているルートテーブルにインターネットゲートウェイ宛のルートの有無)

3.プライベートサブネットに限定できていなければ是正対応

Aose0924 replied:
ご返信ありがとうございます。
ご教示いただいた通り、サブネットグループに含まれるサブネットをプライベートサブネットに限定するように修正および、lecture04.mdの『RDSのサブネットグループがプライベートサブネットに限定されていることを確認』箇所を修正しましたので内容のご確認をお願いします。

dai-fuji replied:
２つに絞っていただいたサブネットがプライベートサブネットであるエビデンスがないと第三者からはプライベートサブネットであるかどうかが分かりませんので、インターネットゲートウェイへのルーティングがないことがわかるエビデンスを追加お願いします！

dai-fuji reviewed (commented):
コメントしましたので確認お願いします！

Aose0924 committed:
Add files via upload

Aose0924 committed:
Add files via upload

Aose0924 committed:
Update lecture04.md

dai-fuji commented on lecture04.md:
```diff
@@ -0,0 +1,91 @@
+![RDS_0407-8](/RDS_0407-8.PNG)
+![RDS_0407-9](/RDS_0407-9.PNG)
+#### RDSのVPCをraisetech-vpc (vpc-041b64c68c22b15c7)へ変更
+![RDS_0418-1](/RDS_0418-1.PNG)
```
こちらの画像ではまだサブネットが4つ含まれています！
画像の差し替え漏れでしょうか？

Aose0924 replied:
差し替えました。失礼しました

dai-fuji reviewed (commented):
もう一息ですね！
コメント確認お願いします🤲

Aose0924 committed:
Add files via upload

Aose0924 committed:
Update lecture04.md

Aose0924 commented:
エビデンスを追加しました。内容のご確認をお願いします。

dai-fuji reviewed (commented):
RDSとしてはOKです！
ただ、パブリックサブネットが無くなってしまっているように思いますが、当初配置されていたEC2およびEC2へのSSH接続はどうなっていますでしょうか？？

Aose0924 commented:
すみません。ご指摘の通りSSH接続できないですね…
ローカルからteratermでSSH接続する場合、やはりパブリックサブネットとインターネットGWの設定が必要という認識でよいですか？
SSH用にエンドポイントを作成しEC2 Instance Connect エンドポイントを使用してSSH接続を試みようとしたのですが接続できず。そもそもローカルからteraterm接続ではないのですが…

dai-fuji commented:
EC2への接続方法はSSHではなくても、TerratermでなくてもOKです！

> ローカルからteratermでSSH接続する場合、やはりパブリックサブネットとインターネットGWの設定が必要という認識でよいですか？

また、AWSの仕様に基づくような内容でYES or NOの回答が求められるような質問にはそのまま答えづらいので（受講生さんの成長機会を奪ってしまうため）、何か調べてトライしたけど上手くいかないなどあれば、質問相談所を利用してもらうと嬉しいです！

Aose0924 committed:
Add files via upload

Aose0924 committed:
Update lecture04.md

プライベートサブネットのみのネットワーク設定のため、Session ManagerにてEC2インスタンス接続、RDS接続確認を行いました。
SSH以外でRDS接続確認ができているかの観点でレビューをお願いします。

Aose0924 committed:
Update lecture04.md

Aose0924 commented:
プライベートサブネットのみのネットワーク設定のため、前回のSSH接続ができず、Session ManagerによるEC2インスタンス接続、RDS接続確認を行いました。
SSH以外でRDS接続確認ができているかの観点でlecture04.mdのレビューをお願いします。

Aose0924 committed:
Update lecture04.md

Aose0924 committed:
Update lecture04.md

dai-fuji commented:
プライベートサブネットを維持してEC2へ接続されたのですね！これは問題なく実現できています！
EC2とRDSの構築およびEC2→RDSへの接続は実現できていますので課題OKです！

なお、今回作成いただいたインターフェースエンドポイントは存在するだけで費用が発生しますのでご注意ください。

第5回の課題ではアプリのデプロイと動作確認が必要になります。その際は現状の構成では実現できないことだ現状認識した上で取り組んでいただくようお願いします！
